juniper/src/validation/rules/limit_number_of_aliases.rs

```diff
@@ -0,0 +1,162 @@
+use crate::{
+    ast::{Field, Operation},
+    parser::Spanning,
+    validation::{ValidatorContext, Visitor},
+    value::ScalarValue,
+};
+
+pub struct Aliases {
+    alias_count: u8,
+    max_allowed: u8,
+}
+
+pub fn factory<'a>() -> Aliases {
+    Aliases {
+        alias_count: 0,
+        max_allowed: 3,
+    }
+}
+
+impl<'a, S> Visitor<'a, S> for Aliases
+where
+    S: ScalarValue,
+{
+    fn enter_operation_definition(
+        &mut self,
+        _ctx: &mut ValidatorContext<'a, S>,
+        _op: &'a Spanning<Operation<'a, S>>,
+    ) {
+        self.alias_count = 0; // Reset for each operation
+    }
+
+    fn enter_field(
+        &mut self,
+        ctx: &mut ValidatorContext<'a, S>,
+        field: &'a Spanning<Field<'a, S>>,
+    ) {
+        let alias_name = &field.item.alias; // Get the Spanning<String> for the alias
+
+        if let Some(alias) = alias_name {
+            self.alias_count += 1;
+            if self.alias_count > self.max_allowed {
+                ctx.report_error(&error_message(&alias.item), &[alias.start]);
+            }
+        }
+    }
+}
+
+fn error_message(alias_name: &str) -> String {
+    format!("There can only be one alias, {} is not allowed", alias_name)
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
Should we have a test confirming that it passes if we pass the max allowed number of aliases?

Reply:
I have added a test case for 3 aliases